### DISCUSSION TITLE
How to control columns width?

### DISCUSSION THREAD

**Discussion by psads-git** [typst/typst#3227] [Q&A] [answered]
The following code produces the output below. As we can see, there is space to accommodate horizontally all 4 items, if we could control the width of each column. That would avoid the overlapping we see in item `c)`. Is there some solution for that? Thanks!

```
#block(
  height: 3.5em,
  columns(4)[
    #enum(
      numbering: "a)",
      indent: 1em,
      enum.item([
        $
        integral^1_0 x dif x.
        $
      ]),

      enum.item([
        $
        integral^1_0  x dif x.
        $
      ]),

      enum.item([
        $
        integral^1_0 x^4 dif x = 2/3," " b=10.
        $
      ]),

      enum.item([
        $
        integral^b_0 x e^x dif x.
        $
      ]),

    )
  ]
)
```

![image](https://github.com/typst/typst/assets/75945439/bdf4552f-8146-4b9e-ba31-a8f9c1c5e4a5)

**Comment by xkevio** [comment, marked as answer]
`columns` is, as the documentation says, for equally sized columns. For other purposes, you would use a `grid` and its `columns` argument which takes a size / array of sizes. By default, it allocates as much space as each element needs.

In your case, we can use a `grid` combined with an `enum` element, like so:
```typ
#set enum(numbering: "a)", indent: 1em)
#let equations = (
  $ integral^1_0 x dif x. $, 
  $ integral^1_0  x dif x. $, 
  $ integral^1_0 x^4 dif x = 2/3," " b=10. $, 
  $ integral^b_0 x e^x dif x. $
)

#block(
  height: 3.5em,
  grid(columns: 4, ..equations.enumerate().map(it => enum(start: it.at(0) + 1, it.at(1))))
)
```

**Reply by psads-git** [reply]
That is perfect, @xkevio! Thanks a lot!
